<!-- Launch in the camera namespace containing "image_raw" and "camera_info" -->
<launch>

  <arg name="manager" /> <!-- Must be globally qualified -->
  <arg name="respawn" default="false" />
  <arg name="left" default="left" />
  <arg name="right" default="right" />
  <arg name="downsampling_factor" value="1" />
  <arg name="approximate_sync" default="false" />

  <arg     if="$(arg respawn)" name="bond" value="" />
  <arg unless="$(arg respawn)" name="bond" value="--no-bond" />

  <!-- Basic processing for left camera -->
  <include file="$(find image_proc)/launch/image_proc.launch"
	   ns="$(arg left)">
    <arg name="manager" value="$(arg manager)" />
    <arg name="respawn" value="$(arg respawn)" />
  </include>

  <!-- Basic processing for right camera -->
  <include file="$(find image_proc)/launch/image_proc.launch"
	   ns="$(arg right)">
    <arg name="manager" value="$(arg manager)" />
    <arg name="respawn" value="$(arg respawn)" />
  </include>

  <!-- Disparity image -->
  <node pkg="nodelet" type="nodelet" name="disparity"
        args="load stereo_image_proc/disparity $(arg manager) $(arg bond)"
	respawn="$(arg respawn)" >
    <param name="approximate_sync" value="$(arg approximate_sync)" />
  </node>

  <!-- PointCloud2 -->
  <node pkg="nodelet" type="nodelet" name="point_cloud2"
        args="load stereo_image_proc/point_cloud2 $(arg manager) $(arg bond)"
	respawn="$(arg respawn)" >
    <param name="approximate_sync" value="$(arg approximate_sync)" />
  </node>
</launch>
